src/local/room_name.rs

```diff
@@ -156,6 +156,19 @@ impl RoomName {
         self.packed
     }
 
+    /// Adds an `(x, y)` pair to this room's name.
+    ///
+    /// # Errors
+    /// Returns an error if the coordinates are outside of the valid room name
+    /// bounds.
+    ///
+    /// For a panicking variant of this function, see [`RoomName::add`].
+    pub fn checked_add(&self, offset: (i32, i32)) -> Result<RoomName, RoomNameParseError> {
+        let (x1, y1) = (self.x_coord(), self.y_coord());
+        let (x2, y2) = offset;
+        Self::from_coords(x1 + x2, y1 + y2)
```

Comment:
If this is really checked, we should be using `x1.checked_add(x2).ok_or(RoomNameParseError)?`.

Reply:
Actually, maybe a bit better in terms of reducing how many `jne`s are needed:
```rust
let (x, y) = x1.checked_add(x2)
     .zip(y1.checked_add(y2))
     .ok_or(RoomNameParseError::TooLarge(...))?;
```

Reply:
Oh wow, completely forgot about overflow when making changes that would prevent overflow in a different way! :laughing:

Reply:
Actually: since `self.AXIS_coord()` is in the range `-128..=127`, adding any `i32` to it cannot overflow around the `i32` space and then back in bounds.  Every time that the function, as currently written, outputs `Ok`, no overflow has occurred, even in intermediate calculations.

Reply:
Right, but as written it panics if either of the coords pairs are close enough to `i32::MIN/MAX` that overflow in that add occurs. Pathological cases, but checked adds shouldn't panic ever.